### PR TITLE
Update installer and permissions so that private links work

### PR DESCRIPTION
**Pull request by asmith030**

### CODE REVIEW

File: modules/cloudaccess/wf_privatelink_manager_definition.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "actions": [
+        "Microsoft.Authorization/roleAssignments/read",
+        "Microsoft.Authorization/roleDefinitions/read",
+        "Microsoft.Network/privateEndpoints/*",
+        "Microsoft.Network/networkInterfaces/*",
+        "Microsoft.Network/privateDnsZones/*",
+        "Microsoft.Network/virtualNetworks/*"
```
Comment (mrsheepuk) [inline modules/cloudaccess/wf_privatelink_manager_definition.json:8]: does this actually need * for this on the virtual networks? that feels like a very wide-ranging permission for what this does...

**Review by mrsheepuk — COMMENTED** [review]
One thought, other than that, this looks fine. As it is off by default I'm comfortable with this going in.

File: wayfinder-cloudaccess.tf [hunk 2]
```diff
@@ -37,6 +38,18 @@ resource "kubectl_manifest" "wayfinder_azure_cloudinfo_cloudaccessconfig" {
...
 
+resource "kubectl_manifest" "wayfinder_azure_privatelinkmanager_cloudaccessconfig" {
+  count      = var.enable_k8s_resources && var.enable_wf_cloudaccess && var.enable_wf_privatelinks ? 1 : 0
+  depends_on = [time_sleep.after_kubectl_manifest_cloud_identity]
+
+  yaml_body = templatefile("${path.module}/manifests/wayfinder-azure-privatelinkmanager-cloudaccessconfig.yml.tpl", {
+    region          = var.location
+    subscription_id = data.azurerm_subscription.current.subscription_id
+    tenant_id       = data.azurerm_subscription.current.tenant_id
+    identity        = "cloudidentity-azure"
```
Comment (KashifSaadat) [inline wayfinder-cloudaccess.tf:49]: small improvement: make this a local and pass to the three or so places that it's referenced.

File: modules/cloudaccess/wf_privatelink_manager_definition.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    "actions": [
+        "Microsoft.Authorization/roleAssignments/read",
+        "Microsoft.Authorization/roleDefinitions/read",
+        "Microsoft.Network/privateEndpoints/read",
+        "Microsoft.Network/privateEndpoints/write",
+        "Microsoft.Network/privateEndpoints/delete",
+        "Microsoft.Network/privateDnsZones/read",
+        "Microsoft.Network/privateDnsZones/write",
```
Comment (KashifSaadat) [inline modules/cloudaccess/wf_privatelink_manager_definition.json:9]: Is `delete` not needed on this resource, for clean-up?

Comment (asmith030) [reply modules/cloudaccess/wf_privatelink_manager_definition.json:9]: It isn't - wayfinder doesn't ever clean up this private DNS zone (by design)

**Review by KashifSaadat — APPROVED** [review]

**Pull request merged**